src/test/java/io/aiven/kafka/connect/gcs/config/GcsSinkConfigTest.java

```diff
@@ -86,6 +88,26 @@ final void incorrectFilenameTemplatesForTopicPartitionRecord(final String templa
         );
     }
 
+    @Test
+    void acceptMultipleParametersWithTheSameName() {
+        final Map<String, String> properties =
+            ImmutableMap.of(
+                GcsSinkConfig.FILE_NAME_TEMPLATE_CONFIG,
+                "{{topic}}-{{timestamp:unit=YYYY}}-"
+                    + "{{timestamp:unit=MM}}-{{timestamp:unit=dd}}"
+                    + "-{{partition}}-{{start_offset:padding=true}}.gz",
+                "gcs.bucket.name", "asdasd"
+            );
+        final Template t = new GcsSinkConfig(properties).getFilenameTemplate();
+        final String fileName = t.instance()
+            .bindVariable("topic", () -> "a")
+            .bindVariable("timestamp", () -> "t")
+            .bindVariable("partition", () -> "p")
+            .bindVariable("start_offset", () -> "c")
+            .render();
+        assertEquals("a-t-t-t-p-c.gz", fileName);
```

Comment:
A proposal for a little more strict checking:
```suggestion
            .bindVariable("timestamp", VariableTemplatePart.Parameter::value)
            .bindVariable("partition", () -> "p")
            .bindVariable("start_offset", () -> "c")
            .render();
        assertEquals("a-YYYY-MM-dd-p-c.gz", fileName);
```

Reply:
then for `start_offset` same thing i will add it

Reply:
fixed